lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -582,7 +582,7 @@ def configure_subplots(self, button):
         toolfig = Figure(figsize=(6, 3))
         canvas = self._get_canvas(toolfig)
         toolfig.subplots_adjust(top=0.9)
-        tool = SubplotTool(self.canvas.figure, toolfig)
+        SubplotTool(self.canvas.figure, toolfig)
```

Comment:
The question this raises is, what keeps the SubplotTool instance alive when no reference to it is saved--not even until the end of this method?  Presumably something external does get a reference, and I just don't see it.  Otherwise, even without this change, the reference would last only until the end of the method.

Reply:
I also don't see who is holding the reference.

Reply:
It's actually a bit curious because the Tool object needs to stay alive even after the function has exited (so that the callbacks are not garbage-collected), but it's not clear to me what makes this possible.
In any case I checked, we do need to maintain a referrence to tool in case e.g. a gc collect()ion occurs in the body of the function, so I reverted this change and left a comment.